data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -251,4 +259,28 @@ static String getPackageVersion() {
         }
         return "";
     }
+    static final int SECONDS_PER_MINUTE = 60;
+    static final int MINUTES_PER_HOUR = 60;
+    static final int HOURS_PER_DAY = 24;
+    static final int SECONDS_PER_HOUR = SECONDS_PER_MINUTE * MINUTES_PER_HOUR;
+    static final int SECONDS_PER_DAY = HOURS_PER_DAY * SECONDS_PER_HOUR;
+
+    public static String formatDurationAsTimespan(Duration duration) {
+        long seconds = duration.getSeconds();
+        int nanos = duration.getNano();
+
+        long hours = (seconds / SECONDS_PER_HOUR) % HOURS_PER_DAY;
```

Comment:
I think I'm missing something here, but as far as I can tell, the logic should be:
`long nanoSecondsRemaining = duration.getNano() + TimeUnit.SECONDS.toNanoseconds(duration.getSeconds());

long days = nanoSecondsRemaining/TimeUnit.DAYS.toNanoseconds(1);
nanoSecondsRemaining -= days * TimeUnit.DAYS.toNanoseconds(1);

int hours = nanoSecondsRemaining/TimeUnit.HOURS.toNanoseconds(1);
nanoSecondsRemaining -= hours * TimeUnit.HOURS.toNanoseconds(1);

int minutes = nanoSecondsRemaining/TimeUnit.MINUTES.toNanoseconds(1);
nanoSecondsRemaining -= minutes * TimeUnit.MINUTES.toNanoseconds(1);

int seconds = nanoSecondsRemaining/TimeUnit.SECONDS.toNanoseconds(1);
nanoSecondsRemaining -= seconds * TimeUnit.SECOND.toNanoseconds(1);
`

Reply:
duration.getNano returns just the remaining nanos as duration is made of (nanos ,seconds)

Reply:
That's why the first line I wrote was:
`duration.getNano() + TimeUnit.SECONDS.toNanoseconds(duration.getSeconds())`

Reply:
i tested mysef